crates/adapters/src/transport/datagen.rs

```diff
@@ -1043,7 +1049,14 @@ impl RecordGenerator {
         obj: &mut Value,
     ) -> AnyResult<()> {
         let min = N::min_value().to_i64().unwrap_or(i64::MIN);
-        let max = N::max_value().to_i64().unwrap_or(i64::MAX);
+        let max = if field.columntype.scale.is_none() {
+            N::max_value().to_i64().unwrap_or(i64::MAX)
```

Comment:
if the scale is specified I need exactly this many, not up to this many

Reply:
see line 408, this is currently a hack because there is no SqlType for u8 so I'm abusing tinyint for generating bytes but tinyint is in range -127..128... I should probably make my own type. The good new is scale is never set for integer types (only decimals) so I'm reusing this field here to indicate I want u8 instead of i8

Reply:
hacks have a tendency to bite you when you expect least